src/oidc-provider.ts

```diff
@@ -0,0 +1,143 @@
+import { Aws, CfnOutput } from 'aws-cdk-lib';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import { Construct } from 'constructs';
+
+/**
+ * Properties for the GitHubActionRole construct.
+ */
+export interface GithubActionRoleProps {
+  /**
+   * A list of GitHub repositories you want to be able to access the IAM role.
+   * Each entry should be your GitHub username and repository passed in as a
+   * single string.
+   *
+   * For example, `['owner/repo1', 'owner/repo2'].
+   */
+  readonly repos: string[];
+
+  /**
+   * The name of the Oidc role.
+   *
+   * @default 'GithubActionRole'
+   */
+  readonly roleName?: string;
+
+  /**
+   * The Github OpenId Connect Provider. Must have provider url
+   * `https://token.actions.githubusercontent.com`. The audience must be
+   * `sts:amazonaws.com`.
+   *
+   * Only one such provider can be defined per account, so if you already
+   * have a provider with the same url, a new provider cannot be created for you.
+   *
+   * @default - a provider is created for you.
+   */
+  readonly provider?: iam.IOpenIdConnectProvider;
+}
+
+/**
+ * Creates or references a Github OIDC provider and accompanying role that trusts the provider.
+ * This role can be used to authenticate against AWS instead of using long-lived AWS user credentials
+ * stored in Github secrets.
+ *
+ * You can do this manually in the console, or create a separate stack that uses this construct.
+ * You must `cdk deploy` once (with your normal AWS credentials) to have this role created for you.
+ *
+ * You can then make note of the role arn in the stack output and send it into the Github Workflow app via
+ * the `githubActionRoleArn` property. The role arn will be `arn:aws:iam::<accountId>:role/GithubActionRole`.
+ *
+ * @see https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services
+ */
+export class GithubActionRole extends Construct {
+  /**
+   * Reference an existing github actions provider.
+   * You do not need to pass in an arn because the arn for such
+   * a provider is always the same.
+   */
+  public static existingGithubActionsProvider(scope: Construct): iam.IOpenIdConnectProvider {
+    return iam.OpenIdConnectProvider.fromOpenIdConnectProviderArn(
+      scope,
+      'GithubActionProvider',
+      `arn:aws:iam::${Aws.ACCOUNT_ID}:oidc-provider/token.actions.githubusercontent.com`,
```

Comment:
I did this, but am curious about a use case where the user is building this construct in an account different from the one they want the github action role to be in (and thus, `Aws.ACCOUNT_ID` will fail).